books/projects/x86isa/machine/instructions/fp/mov.lisp

```diff
@@ -744,6 +744,75 @@
        (x86 (write-*ip proc-mode temp-rip x86)))
     x86))
 
+(def-inst x86-movhlps-sse
+
+  :parents (two-byte-opcodes fp-opcodes)
+
+  :short "Move Packed Single Precision Floating-Point Values High to Low"
+
+  :long
+  "<h3>Op/En = RM: \[OP XMM, XMM\]</h3>
+  NP 0F 12: MOVHLPS xmm1, xmm2<br/>"
+
+  :returns (x86 x86p :hyp (x86p x86))
+
+  :modr/m t
+
+  :body
+
+  (b* (((the (unsigned-byte 4) xmm-index)
+        (reg-index reg rex-byte #.*r*))
+
+       (p2 (prefixes->seg prefixes))
+
+       (p4? (eql #.*addr-size-override*
+                 (prefixes->adr prefixes)))
+
+       (seg-reg (select-segment-register proc-mode p2 p4? mod r/m sib x86))
+
+       (inst-ac? ;; Exceptions Type 5
+        t)
+       ((mv flg0
+            (the (unsigned-byte 128) xmm)
+            (the (integer 0 4) increment-RIP-by)
+            (the (signed-byte 64) ?addr)
+            x86)
+        (x86-operand-from-modr/m-and-sib-bytes proc-mode
```

Comment:
This function reads from memory unless Mod = 11. Does the decoder enforce that? If not, we should read directly from xmm2, obtaining the register index from the R/M bits.

Reply:
Line 2749 of `inst-listing.lisp` requires that Mod = 0x3